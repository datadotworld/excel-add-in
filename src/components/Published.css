/*
 * Copyright 2018 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */

.published-container {
  width: 90%;
  margin: 0 auto;
  padding-top: 30%;
}

.published-title {
  margin: 0;
  text-align: center;
  font-size: 20px;
}

.published-chart {
  margin: 0;
  padding-top: 10px;
  max-width: 100%;
  max-height: 100%;
}

.published-buttons {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: 25px;
}

.published-link {
  color: white;
  width: 47.25%;
}

.published-back-button {
  width: 47.25%;
}

.published-link-button {
  width: 100%;
}
