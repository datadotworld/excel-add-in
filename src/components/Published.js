/*
 * Copyright 2018 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

import './Published.css';
import analytics from '../analytics';

class Published extends Component {
  static propTypes = {
    chart: PropTypes.string,
    uri: PropTypes.string,
    title: PropTypes.string
  }

  componentDidMount () {
    // User will have scrolled to the bottom of the add in when filling in the form
    // Ensure the header is visible when this page is loaded
    window.scrollTo(0, 0)
  }

  viewClicked() {
    analytics.track('exceladdin.published.view_button.click');
  }

  newInsightClicked() {
    analytics.track('exceladdin.published.new_button.click');
    window.location.pathname = '/insights'
  }

  render() {
    const { title, chart, uri } = this.props;
    return (
      <div className="published">
        <div className="published-container">
          <h3 className="published-title">{`${title} was published!`}</h3>
          <img
            className="published-chart"
            src={`data:image/png;base64, ${chart}`}
            alt="chart"
          />

          <div className='published-buttons'>
            <Button
              className="published-back-button"
              bsSize="xsmall"
              onClick={this.newInsightClicked}
            >
              Publish new insight
            </Button>
            <a className="published-link" href={uri} target="_blank" rel="noopener noreferrer">
              <Button
                className="published-link-button"
                bsStyle='primary'
                bsSize="xsmall"
                onClick={this.viewClicked}
              >
                View on data.world
              </Button>
            </a>
          </div>
        </div>
      </div>
    );
  }
}

export default Published;
