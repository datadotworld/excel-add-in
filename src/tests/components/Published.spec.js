/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

import Published from '../../components/Published';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Published />, div);
});

it('renders published insight', () => {

  expect(renderer.create(<Published
    chart={'b64ImageString'}
    uri={'https://data.world/path/to/insight'}
  />).toJSON()).toMatchSnapshot()
});
