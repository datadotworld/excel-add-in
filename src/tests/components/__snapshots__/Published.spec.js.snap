// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders published insight 1`] = `
<div
  className="published"
>
  <div
    className="published-container"
  >
    <h3
      className="published-title"
    >
      undefined was published!
    </h3>
    <img
      alt="chart"
      className="published-chart"
      src="data:image/png;base64, b64ImageString"
    />
    <div
      className="published-buttons"
    >
      <button
        className="published-back-button btn btn-xs btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Publish new insight
      </button>
      <a
        className="published-link"
        href="https://data.world/path/to/insight"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          className="published-link-button btn btn-xs btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          View on data.world
        </button>
      </a>
    </div>
  </div>
</div>
`;
